### DISCUSSION TITLE
Using Github Projects, in particular the Iteration template

### DISCUSSION THREAD

**Discussion by Niravanaa** [NathanGrenier/SOEN-341#12] [Ideas]
We should use the `Iteration`-related template in GitHub projects, and have maybe a little guide oriented around how to use it, particularly in terms of what the team member would be viewing and how to make use of the filter box to further navigate the issues that will be created during the sprints.

We'd have a `Backlog` view that houses all the issues created, both categorized and not categorized. 
A `Current Iteration (Sprint)` view containing the five Issue statuses (Backlog, In Progress, In Review, Done). 
A `Next Iteration (Sprint)` view containing the issue items that will tackled in the next sprint.
A `Roadmap` view containing a calendar-view of the current issues being tackled with their expected start and end dates.
An `In Review` view containing the Issue items for which pull requests are currently being reviewed.
Finally, a `My Items` view that filters the issues to display issue items that the user in question must currently/later work on. 

Here is an example of the views, provided by GitHub upon creating a project:
![memex-templates-iterative-development-dark-c83add2b3d99](https://github.com/NathanGrenier/SOEN-341/assets/64038962/b135b121-e3e9-46ac-b88b-e0c0459890a3)

**Comment by NathanGrenier** [comment]
A github project already exists in the repo. I haven't fully set it up, but plan to in the future.
The different tabs will have to be configured properly. We can also add any other tabs that we'll need for project management.

Here's a link to github documentation about projects: https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects

![image](https://github.com/NathanGrenier/SOEN-341/assets/56941028/86c837bd-fb04-4215-9897-f6f93467d38c)

**Reply by Niravanaa** [reply]
I see! It's because you haven't linked it to the repo, that's why I don't see it yet.

**Reply by NathanGrenier** [reply]
I'm pretty sure it is linked, check again.

**Reply by Niravanaa** [reply]
Awesome, I see it now. Thanks

**Reply by NathanGrenier** [reply]
ya, I changed the visibility to public, but it should have also been working before because I've added everyone to the project.

**Comment by NathanGrenier** [comment]
@Niravanaa Are you interested in setting up the project page?

**Reply by Niravanaa** [reply]
Yes, sure

The current setup looks good, just that I would add a `Next Iteration` tab that will contain the issue items that we may want to work on during the next sprint and not the current one.

**Reply by NathanGrenier** [reply]
One other thing I know you'd have to set up is the iteration cycles (length, start and end date). Other than that, there might be some extra features to set up that you can read about in the GitHub projects documentation.

I also made an issue for the task.
